static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -57,10 +59,33 @@ export default function DataSecrecy() {
         data: {allowSuperuserAccess: value},
       });
       setAllowAccess(value);
-      addSuccessMessage(t('Successfully updated access.'));
+
+      // if the user has allowed access, we need to remove the temporary access window
+      // only if there is an existing waiver
+      if (value && allowData) {
+        await api.requestPromise(`/organizations/${organization.slug}/data-secrecy/`, {
+          method: 'DELETE',
+        });
+        setAllowDateFormData('');
+        setAllowDate(undefined);
+        addSuccessMessage(
+          t('Successfully removed temporary access window and allowed support access.')
+        );
+        // refetch to get the latest waiver data
+        refetch();
+        return;
+      }
+      addSuccessMessage(
+        value
+          ? t('Successfully allowed support access.')
+          : t('Successfully removed support access.')
+      );
```

Comment:
nit: instead of early return you can modify the success message text, add the success message outside of the `if`, and then you can use the `refetch` at the end of the function